src/components/LazyLog/index.jsx

```diff
@@ -525,10 +618,54 @@ export default class LazyLog extends Component {
     }
   };
 
-  handleFormatPart = () => {
-    const { isSearching, searchKeywords } = this.state;
+  handleFormatPart = lineNumber => {
+    const {
+      isSearching,
+      searchKeywords,
+      resultLines,
+      currentResultsPosition,
+    } = this.state;
+    const { searchLikeBrowser } = this.props;
 
     if (isSearching) {
+      // If browser-search has started and we're on the line
+      // that has the search term that is selected
+      if (
+        searchLikeBrowser &&
+        resultLines &&
+        currentResultsPosition !== undefined &&
+        resultLines[currentResultsPosition] === lineNumber
+      ) {
+        // This finds which word in the line should be the highlighted one.
+        // For example, if we should be highlighting the 2nd match on line 18,
+        // this would set locationInLine to 2.
+        let locationInLine = 0;
```

Comment:
This seems to blow up the farther down the text we get, is it possible to optimize? I also don't quite think I understand how this works.

Reply:
Yeah it's kinda weird and confusing. I added comments and made it much quicker.